test/unit/auth.test.js

```diff
@@ -33,80 +36,98 @@ describe('auth', () => {
     ]);
   });
 
-  it('returns ncurc data if it is present and valid', () => {
-    runAuthScript({ username: 'nyancat', token: '0123456789abcdef' }, [
+  it('returns ncurc data if it is present and valid', async function() {
+    this.timeout(1000);
+    await runAuthScript({ username: 'nyancat', token: '0123456789abcdef' }, [
       'bnlhbmNhdDowMTIzNDU2Nzg5YWJjZGVm'
     ]);
   });
 });
 
 function runAuthScript(ncurc = undefined, expect = []) {
-  const HOME = path.resolve(__dirname, `tmp-${testCounter++}`);
-  rimraf.sync(HOME);
-  mkdirp.sync(HOME);
-  const ncurcPath = path.resolve(HOME, '.ncurc');
+  return new Promise((resolve, reject) => {
+    const HOME = path.resolve(__dirname, `tmp-${testCounter++}`);
+    rimraf.sync(HOME);
+    mkdirp.sync(HOME);
+    const ncurcPath = path.resolve(HOME, '.ncurc');
 
-  if (ncurc !== undefined) {
-    if (typeof ncurc === 'string') {
-      fs.writeFileSync(ncurcPath, ncurc, 'utf8');
-    } else {
-      fs.writeFileSync(ncurcPath, JSON.stringify(ncurc), 'utf8');
+    if (ncurc !== undefined) {
+      if (typeof ncurc === 'string') {
+        fs.writeFileSync(ncurcPath, ncurc, 'utf8');
+      } else {
+        fs.writeFileSync(ncurcPath, JSON.stringify(ncurc), 'utf8');
+      }
     }
-  }
 
-  const proc = spawn(process.execPath,
-    [ require.resolve('../fixtures/run-auth') ],
-    // XXX this could just be env: { ...process.env, HOME } but the test loader
-    // is complaining?
-    { env: Object.assign({}, process.env, { HOME }) });
-  let stderr = '';
-  proc.stderr.setEncoding('utf8');
-  proc.stderr.on('data', (chunk) => { stderr += chunk; });
-  proc.on('close', () => {
-    if (stderr) { throw new Error(`unexpected stderr:\n${stderr}`); }
-  });
+    const proc = spawn(process.execPath,
+      [ require.resolve('../fixtures/run-auth') ],
+      // XXX this could just be env: { ...process.env, HOME } but the test loader
```

Comment:
Ah we already know why this happens in https://github.com/joyeecheung/node-core-utils/pull/12#issuecomment-339955478 ...Can remove this now.

Reply:
Yup… it’s a shame though :smile:

Reply:
@addaleax We'll just have to wait until 2018 (March?), I guess

Reply:
Is that the acorn release schedule or why March? ^^

Reply:
like, this works in Node 8 & 9 without a flag, I’m surprised userland parsers haven’t picked it up before V8…

Reply:
@addaleax IIRC ECMA has an assembly that approve the spec in March so by Feb they would need to cut releases before submitting them.

Reply:
@addaleax Acorn has a few plugins that support the object spread syntax already, they are just not included by default, I guess they are waiting for the spec to drop.